Add blog post on beginner's guide to app sec (#1118)
oktadeveloper/okta-blog: _source/_posts/2022-04-08-beginners-app-sec.md

As attackers target applications ever more aggressively, keeping them secure plays a vital role in a holistic cybersecurity strategy. Application security is necessary to avoid financial and legal repercussions, protect your organization's reputation, and build trust with your customers and partners.

## Application security concepts

There are a few concepts you should understand before you develop an application security strategy. Learning these will help you avoid common pitfalls or ineffective techniques that could compromise the security of your application.

### OWASP top 10

> The [Open Web Application Security Project (OWASP)](https://en.wikipedia.org/wiki/OWASP) online community studies and reports on web application security. The [OWASP Top 10 report lists the most critical security vulnerabilities an application can face.

Reviewer: ```suggestion
The [Open Web Application Security Project (OWASP)](https://owasp.org/) online community studies and reports on web application security. The [OWASP Top 10](https://owasp.org/www-project-top-ten/) report lists the most critical security vulnerabilities an application can face.

```

The list—based on survey data from cybersecurity professionals—is determined by the frequency and likelihood of a vulnerability, how easy it is to detect, and the impact it can have on your organization.

OWASP's report can help you assess areas of your application that present higher potential risk. It provides example attack scenarios to show how various vulnerabilities work, as well as guidelines for avoiding these attacks.

Below are the top ten security vulnerabilities as noted in the latest [OWASP report](https://owasp.org/www-project-top-ten/) from 2021:

1. Broken access control
2. Cryptographic failures
3. Injection
4. Insecure design
5. Security misconfiguration
6. Vulnerable and outdated components
7. Identification and authentication failures
8. Software and data integrity failures
9. Security logging and monitoring failures
10. Server-side request forgery

>  ### Authentication basics

Reviewer: ```suggestion
### Authentication basics

```

---

> A Beginner's Guide to  Application Security

Author: ```suggestion

```

Application security, or *AppSec*, as it's commonly known, takes place throughout the development lifecycle. It's the process of creating applications that are secure from both internal and external threats. Internal threats can range from simple human error to malicious acts. External threats, which include data breaches, malware, and phishing attacks, can be constant and costly. It's estimated that distributed denial-of-service (DDoS) attacks alone [will grow to 15.4 million by 2023](https://www.cisco.com/c/en/us/solutions/collateral/executive-perspectives/annual-internet-report/white-paper-c11-741490.html).

As attackers target applications ever more aggressively, keeping them secure plays a vital role in a holistic cybersecurity strategy. Application security is necessary to avoid financial and legal repercussions, protect your organization's reputation, and build trust with your customers and partners.

## Application security concepts

There are a few concepts you should understand before you develop an application security strategy. Learning these will help you avoid common pitfalls or ineffective techniques that could compromise the security of your application.

> ### OWASP top 10

Author: ```suggestion
### OWASP Top Ten 

```
because it's the name of a thing, and looks weird in lowercase.

You might be inspired to learn even more about application security. If that's the case, consider pursuing [certification](https://www.csoonline.com/article/3631530/8-top-cloud-security-certifications.html), so you can develop expertise in cloud security. This in-demand skill set will benefit you as well as your company or organization. 

## Use your knowledge

Application security needs to be a top priority for developers. Modern applications often run on multiple networks and connect to the cloud, increasing the number of potentially problematic access points for attackers.

Many organizations are familiar with security best practices but may not understand the concepts behind them. Using this knowledge, you can implement the solutions described above to achieve great results. 

Check out these posts for more information about application security:

* [Webinar: Securing Remote Access: The Intersection of Identity and Network Security](https://www.okta.com/resources/webinar-securing-remote-access-the-intersection-of-identity-and-network-security/)

* [A Brief History of Zero Trust Security](/blog/2018/08/a-brief-history-of-zero-trust-security/)

> * You can search a collection of recent Okta Developer blog posts that cover [aspects of app security](​​https://developer.okta.com/search/#q=app%20security) or check out this free book about [API security](https://developer.okta.com/books/api-security/) written by members of the Okta Dev Advocacy team back in 2019. 

Author: ```suggestion
* You can search a collection of recent Okta Developer blog posts that cover [aspects of app security](​​https://developer.okta.com/search/#q=app%20security&f:@commonoktasource=[Developer%20blog]) or check out this free book about [API security](https://developer.okta.com/books/api-security/) written by members of the Okta Dev Advocacy team back in 2019. 

```
updated url limits search to the dev blog
